redfish-core/lib/assembly.hpp

```diff
@@ -51,6 +51,44 @@ inline void
         tempyArray.at(assemblyIndex)["Status"]["Health"] = "OK";
         tempyArray.at(assemblyIndex)["Status"]["State"] = "Enabled";
 
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "EBMC")
+        {
+            auto bus = sdbusplus::bus::new_default();
+            constexpr auto adcSensorBatteryPath =
+                           "/xyz/openbmc_project/sensors/voltage/Battery_Voltage";
+
+            auto method = bus.new_method_call("xyz.openbmc_project.ObjectMapper",
+                                              "/xyz/openbmc_project/object_mapper",
+                                              "xyz.openbmc_project.ObjectMapper",
+                                               "GetObject");
+            method.append(adcSensorBatteryPath);
+            std::vector<std::string> interface;
+            method.append(interface);
+
+            std::vector<std::pair<std::string, std::vector<std::string>>> response;
+
+            try
```

Comment:
No need for a try/catch if you follow  crow::connections::systemBus->async_method_call(